src/wallet/wallet2.cpp

```diff
@@ -13046,6 +13055,77 @@ void wallet2::throw_on_rpc_response_error(const boost::optional<std::string> &st
   THROW_WALLET_EXCEPTION_IF(*status != CORE_RPC_STATUS_OK, tools::error::wallet_generic_rpc_error, method, m_trusted_daemon ? *status : "daemon error");
 }
 //----------------------------------------------------------------------------------------------------
+
+bool wallet2::save_to_file(const std::string& path_to_file, const std::string& raw, bool is_hex) const
+{
+  if (is_hex || m_export_format == ExportFormat::Raw)
+  {
+    return epee::file_io_utils::save_string_to_file(path_to_file, raw);
+  }
+
+  std::string h = HEX_OUTPUT_MAGIC + "\n";
+  h.append(epee::string_tools::buff_to_hex_nodelimer(raw));
+
+  // Split the long string into multiple lines, similarly to how openssl
+  // prints certificates.
+  for (size_t i = 63; i < h.length(); i += 64)
+  {
+    h.insert(i, 1, '\n');
+  }
+
+  h.insert(h.end(), '\n');
+
+  return epee::file_io_utils::save_string_to_file(path_to_file, h);
+}
+//----------------------------------------------------------------------------------------------------
+
+bool wallet2::load_from_file(const std::string& path_to_file, std::string& target_str,
+                             size_t max_size)
+{
+  std::string data;
+  bool r = epee::file_io_utils::load_file_to_string(path_to_file, data, max_size);
+  if (!r)
+  {
+    return false;
+  }
+
+  // Ignore any whitespace.
+  // We try to convert the version without whitespace from hex to binary.
+  // If that fails, we stick to the original data (hence the copy).
+  std::string without_whitespace(data);
+  without_whitespace.erase(
+    std::remove_if(
+      without_whitespace.begin(),
+      without_whitespace.end(),
+      [](char c){ return (c =='\r' || c =='\t' || c == ' ' || c == '\n'); }),
+    without_whitespace.end());
+
+  if (0 != without_whitespace.compare(0, HEX_OUTPUT_MAGIC.length(), HEX_OUTPUT_MAGIC))
+  {
+    // It's NOT our hex dump.
+    target_str = std::move(data);
+    return true;
+  }
+  // It's apparently our hex dump.
+
+  without_whitespace.erase(0, HEX_OUTPUT_MAGIC.length());
+  r = epee::string_tools::validate_hex(without_whitespace.size(), without_whitespace);
```

Comment:
The hex string parsing should be doing a validation too (making this unnecessary) ?

Reply:
Calling `validate_hex` makes it faster to fail, a pure usability improvement (provided the difference is noticeable). Not sure if it's worth complicating the code for this, but it seems easy enough. If you insist on removing this, I don't have a problem doing it.

Reply:
With base58 this is going away regardless.